### PR TITLE
 feat/roe-2725: retrieve application data from the DB for registration middleware

### PR DESCRIPTION
**Pull request by mwejuli-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2725

### Change description

- Retrieves application data from the DB for registration middleware
- Modifies tests to match new/updated functionality
- Includes minor formatting changes

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

**Review by lduranteau — APPROVED** [review]

**Comment by lduranteau** [comment]
1 code smell

File: test/controllers/trust.information.controller.spec.ts [hunk 1]
```diff
@@ -163,8 +193,9 @@ describe("TRUST INFORMATION controller", () => {
     });
 
     test("catch error when rendering the page", async () => {
-      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
-      mockGetApplicationData.mockImplementationOnce(() => { throw ERROR; });
+      // mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
```
Comment (markpit) [inline test/controllers/trust.information.controller.spec.ts:196]: this comment can be removed if not needed

Comment (mwejuli-ch) [reply test/controllers/trust.information.controller.spec.ts]: removed.

**Comment by mwejuli-ch** [comment]
> 1 code smell

code smell resolved

**Pull request merged**